src/Spago/Command/Fetch.purs

```diff
@@ -274,66 +279,68 @@ run { packages: packagesRequestedToInstall, ensureRanges, isTest, isRepl } = do
       Right _lockfile -> pure dependencies
       Left reason -> writeNewLockfile reason dependencies
 
-type LockfileBuilderResult =
-  { workspacePackages :: Map PackageName Lock.WorkspaceLockPackage
-  , packages :: Map PackageName Lock.LockEntry
-  }
-
 writeNewLockfile :: forall a. String -> PackageTransitiveDeps -> Spago (FetchEnv a) PackageTransitiveDeps
 writeNewLockfile reason allTransitiveDeps = do
```

Comment:
I took the liberty to reorganize a bit how this function worked. It took me quite a while to understand the meaning, and once I did, it occurred to me that the implementation was needlessly convoluted. What it's really doing is separating "workspace packages" from "non-workspace packages" and then converting each set to their respective types of lockfile entry.

Reply:
Good point! I didn't mind the previous shape, but I don't mind the new one either, so all good 🙂